src/components/nodes/boundaryEvent/boundaryEvent.vue

```diff
@@ -10,10 +10,43 @@ import { id as taskId } from '@/components/nodes/task';
 import { id as callActivityId } from '@/components/nodes/callActivity';
 import { id as manualTaskId } from '@/components/nodes/manualTask';
 import { id as scriptTaskId } from '@/components/nodes/scriptTask';
+import portsConfig from '@/mixins/portsConfig';
+import { portGroups } from '@/mixins/portsConfig';
+import { g } from 'jointjs';
+
+function getPointFromGroup(model, group) {
+  const { x: shapeX, y: shapeY } = model.position();
+  const { x, y } = Object.values(model.getPortsPositions(group))[0];
+  return new g.Point(shapeX + x, shapeY + y);
+}
+
+function getPortPoints(model) {
+  return portGroups.filter(group => group !== 'absolute').map(group => getPointFromGroup(model, group));
+}
+
+function closestPort(model, anchorReference) {
+  return getPortPoints(model).sort((p1, p2) => {
+    const referencePoint = new g.Point(anchorReference.x, anchorReference.y);
+    return referencePoint.distance(p1) - referencePoint.distance(p2);
+  })[0];
+}
+
+function hasPorts(model) {
+  return Object.values(model.getPortsPositions(portGroups[0])).length > 0;
+}
+
+function snapToAnchor(coords, model) {
```

Comment:
This is the one name in this group that I don't like. This function doesn't snap the event to an anchor but it returns a point that the event should be anchored to. Can you think of a better name for this?

Reply:
I tried getAnchorCoordinates. But maybe there is something better.